velox/docs/velox-in-10-min.rst

```diff
@@ -0,0 +1,321 @@
+===================
+Velox in 10 minutes
+===================
+
+This is a quick introduction into Velox, the new C++ vectorized database
+acceleration library aimed at optimizing query engines and data processing
+systems. This tutorial was inspired by `TorchArrow in 10 minutes
+<https://github.com/facebookresearch/torcharrow/blob/main/tutorial/tutorial.ipynb>`_
+which itself was based on a `10 minutes to pandas tutorial <https://pandas.pydata.org/docs/user_guide/10min.html>`_.
+
+We are going to briefly touch all major components starting with data
+representation, followed by expression evaluation, followed by aggregations,
+sorting and joins.
+
+Note: Velox doesn't provide SQL interface, but in this tutorial to keep
+things simple we are going to use SQL and utilities that were designed
+to simplify test development. These utilities use `DuckDB <https://duckdb.org/>`_
+to parse SQL.
+
+The code used in this tutorial is available in `velox/exec/tests/VeloxIn10MinDemo.cpp <https://github.com/facebookincubator/velox/blob/main/velox/exec/tests/VeloxIn10MinDemo.cpp>`_.
+
+Let’s get started.
+
+Vectors
+-------
+
+Velox uses columnar representation of the data similar to Arrow called vectors.
+Let’s create two vectors to store 64-bit integers.
+
+.. code-block:: c++
+
+    auto a = makeFlatVector<int64_t>({0, 1, 2, 3, 4, 5});
+    auto b = makeFlatVector<int64_t>({0, 5, 10, 15, 20, 25});
```

Comment:
An example with Strings will be nice.

Reply:
I see that a `Filter` usage is missing. We could use the string vector to add a filter.